src/stylesheets/core/_deprecated.scss

```diff
@@ -0,0 +1,52 @@
+/*  deprecated.scss
+    ---
+    Occasionally the design system will deprecate
+    old variables or functionality. If we replace
+    the old functionality with something new, this is a
+    place to connect the old functionality to the
+    new functionality, in the service of better
+    continuity and backwards compatibility within a
+    major release cycle.
+
+    Note the USWDS version where we deprecated the
+    old functionality in a comment.
+
+    Add the deprecation to the warning.
+
+    This file should started fresh at each
+    major version.
+*/
+
+@if $theme-show-deprecation-warnings {
+  @warn "\a"
+    + "--------------------------------------------------------------------\a"
+    + "Deprecated in USWDS 2.x\a"
```

Comment:
Maybe this should be more like `Notifications` and less specifically about deprecations